data_validation/combiner.py

```diff
@@ -34,27 +36,106 @@
+    wrapper simply manages the input columns and stitches the results back together.
+    This is because validations of > 140(ish) columns trigger a RecursionError when unioning Ibis subqueries.
+    In this method we pass in column slices of the incoming Dataframes and combine the results.
+
+    It is a bit of a hack but I cannot find a way to optimize the Ibis processing. It appears to be
+    inefficient in that we create a subquery for each validation (column) in _calculate_differences() and
+    then union them all. We then do the same on the source/target table expressions to join it all back
+    together again. I (nj1973) spent a singificant amount of time trying to understand/optimize the Ibis
+    processing but fell back on this simpler (less risky) workaround.
+
+    Returns:
+        pandas.DataFrame:
+            A pandas DataFrame with the results of the validation in the same
+            schema as the report table.
+    """
+    _check_schema_names(source_df, target_df)
+
+    join_on_fields = tuple(join_on_fields)
+
+    validation_columns = run_metadata.validations.keys()
+    slice_thresholds = [
+        (_ * COMBINER_COLUMN_SLICE_WIDTH)
+        for _ in range(int(len(validation_columns) / COMBINER_COLUMN_SLICE_WIDTH) + 1)
+    ]
+
+    result_df = None
+    # Process the input Dataframes in slices of columns to avoid "RecursionError"s.
+    for slice_start in slice_thresholds:
+        columns_in_vertical_slice = list(validation_columns)[
+            slice_start : slice_start + COMBINER_COLUMN_SLICE_WIDTH
+        ]
+        # Ensure any join columns are included in the column slice.
+        columns_in_vertical_slice.extend(
+            set(join_on_fields) - set(columns_in_vertical_slice)
+        )
+        # Ensure join_on_fields columns are in the slice.
+        interim_result_df = _generate_report(
+            run_metadata,
+            source_df[columns_in_vertical_slice],
+            target_df[columns_in_vertical_slice],
+            join_on_fields=join_on_fields,
+            is_value_comparison=is_value_comparison,
+            verbose=verbose,
+        )
+        if result_df is None:
+            result_df = interim_result_df
+        else:
+            result_df = pandas.concat([result_df, interim_result_df])
+
+    # Get the first validation metadata object to fill source and/or target empty table names.
+    first = run_metadata.validations[next(iter(run_metadata.validations))]
+    if first.validation_type != consts.CUSTOM_QUERY:
+        result_df.source_table_name.fillna(
+            first.get_table_name(consts.RESULT_TYPE_SOURCE), inplace=True
+        )
+        result_df.target_table_name.fillna(
+            first.get_table_name(consts.RESULT_TYPE_TARGET), inplace=True
+        )
+
+    _get_summary(run_metadata, result_df, source_df, target_df)
+
+    return result_df
+
+
+def _generate_report(
```

Comment:
```suggestion
def _generate_report_item(
```
Just an idea for a new name, you can use something else you feel is more appropriate.